src/utils/swisseph/utils.swisseph.ts

```diff
@@ -4,169 +4,95 @@ import { Planet } from '../../types/types';
-
-          const planet:Planet = {
-            name: `mars`,
-            sign: `${signos[getIndex(body.longitude)]}`,
-            deg: parseInt(`${getDegree(body.longitude)}`, 10),
-            min: parseInt(`${getMin(body.longitude, getDegree(body.longitude))}`, 10)
-
-          }
-
-          planetIsValid(planet) ? planets.push(planet) : null 
-
-          
-        })
-
-        // Jupiter position
-        swisseph.swe_calc_ut (julday_ut, swisseph.SE_JUPITER, flag, function (body:any) {
-
-          //__res += `Jupiter en ${signos[getIndex(body.longitude)]}, en el grado ${getDegree(body.longitude)}, y el minuto ${getMin(body.longitude, getDegree(body.longitude))} n`
-
-          const planet:Planet = {
-            name: `jupiter`,
-            sign: `${signos[getIndex(body.longitude)]}`,
-            deg: parseInt(`${getDegree(body.longitude)}`, 10),
-            min: parseInt(`${getMin(body.longitude, getDegree(body.longitude))}`, 10)
+buildPlanet = (_index:number, _julday_ut:any) => {
+    
+    /**
+     * ?
+     * ? passing _index as SE_#PLANET#:number to swisseph.swe_cal_ut since they are actually just numbers and share index from our planets_names array structure ["sun"... "saturn"] [0 ... 6]
+     * ?
+     * ? see node_modules/swisseph/lib/swisseph.js lines 16 to 28
+     * ?
+     */
+
+    swisseph.swe_calc_ut (_julday_ut, _index, flag, function (body:any) {
+
+      const planet:Planet = {
+        name: `${planets_names[_index]}`,
+        sign: `${sign_names[getIndex(body.longitude)]}`,
+        deg: parseInt(`${getDegree(body.longitude)}`, 10),
+        min: parseInt(`${getMin(body.longitude, getDegree(body.longitude))}`, 10)
 
-          }
+      }
+    
+      planetIsValid(planet) ? planets.push(planet) : null 
 
-          planetIsValid(planet) ? planets.push(planet) : null 
+    })
 
-          
-        })
+}
 
-        // Saturn position
-        swisseph.swe_calc_ut (julday_ut, swisseph.SE_SATURN, flag, function (body:any) {
+// path to ephemeris data
+swisseph.swe_set_ephe_path (__dirname + '/../ephe')
 
-         // __res += `Saturno en ${signos[getIndex(body.longitude)]}, en el grado ${getDegree(body.longitude)}, y el minuto ${getMin(body.longitude, getDegree(body.longitude))}`
-          const planet:Planet = {
-              name: `saturn`,
-              sign: `${signos[getIndex(body.longitude)]}`,
-              deg: parseInt(`${getDegree(body.longitude)}`, 10),
-              min: parseInt(`${getMin(body.longitude, getDegree(body.longitude))}`, 10)
 
-            }
+export const allPlanets = (_year:string, _month:string, _day:string, _hour:number | never):Planet[] | [] => {
 
-            planetIsValid(planet) ? planets.push(planet) : null 
+    const _pos = swisseph.swe_julday(parseInt(_year, 10), parseInt(_month, 10), parseInt(_day, 10), _hour, swisseph.SE_GREG_CAL, (julday_ut:any) => {
+        
+        for (let _i:number = 0; _i < planets_names.length; _i++) {          
```

Comment:
```suggestion
       planets.forEach(planet => calculatePosition(planet, jul_day))
```

O algo asi..que te parece?

Reply:
Me parece guay pero no acabo de entender que estás iterando en el forEach. Yo lo que uso en el for loop es planets_names que es nuestro propio array de planetas. 

Lo que nos interesa es pasar el index asociado a ese planeta desde planets_names a la función buildPlanet para usar la librería swe sin tener que referenciar cada uno de los planetas con la constante que trae la librería (SE_SUN, SE_MOON, etc...). O sea sun es 0, moon es 1... 

Igual estas abstrayendo el calculo desde `buildPlanets()` a una nueva función `calculatePosition()` ?